sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1257,6 +1257,12 @@ void ProgramManager::addImages(pi_device_binaries DeviceBinary) {
       if (KSIdIt != KSIdMap.end()) {
         auto &Imgs = m_DeviceImages[KSIdIt->second];
         assert(Imgs && "Device image vector should have been already created");
+        // If the images differ in target format, the dumping is necessary.
+        if (DumpImages &&
+            !std::all_of(Imgs->begin(), Imgs->end(), [&](auto &I) {
+              return I->getFormat() == Img->getFormat();
+            }))
```

Comment:
Is there any case where this wouldn't hold, that is we add another image with the same format to the kernel-set?

Reply:
I was prompted by [PI_DEVICE_BINARY_TYPE_LLVMIR_BITCODE](https://github.com/intel/llvm/blob/08d281537f89029f4113b6ecfffeaf5a6ed5cf2a/sycl/include/sycl/detail/pi.h#L750) and though that there could, hypothetically, be more than one target that would compile down to bitcode, and then dumping only one should be sufficient. Happy to remove that check and always call to `dumpImage`.

Reply:
I would hope that all images would be different in some way, so I figure dumping all should be fine, but I am okay with either.

Reply:
@steffenlarsen I've changed it to always dump images, regardless of their type, the slight annoyance is that for known kernels it will use a sticky sequence ID to prevent overriding entries. The proper way of solving it would be to have `ProgramManager` track what has already been dumped, something along the line of:
```cpp
using DumpsPerBinaryType =
    std::unordered_map<RT::PiDeviceBinaryType, uint32_t>;
std::unordered_map<std::string, DumpsPerBinaryType> m_ImageDumps;
```
but it seems like an overkill for this.